tool/src/kotlin/mod.rs

```diff
@@ -401,51 +564,196 @@ return string"#
             return_type_name,
             borrows,
             cleanups,
+            val_name,
+            return_type_modifier,
         };
         opaque_return
             .render()
             .expect("Failed to render opaque return block")
     }
 
-    fn gen_infallible_return<'d>(
+    fn gen_fallible_return<'d>(
```

Comment:
nit: this generates the return type _conversion_, yes? perhaps we should have that in the name (gen_fallible_return_conversion or something). same for other methods